lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8794,23 +8796,40 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
     // FIXME: Hard-code @MainActor and @UIActor, because we don't have a
     // point at which to do name lookup for imported entities.
     if (isMainActorAttr(swiftAttr)) {
-      if (SeenMainActorAttr) {
+      if (seenMainActorAttr) {
         // Cannot add main actor annotation twice. We'll keep the first
         // one and raise a warning about the duplicate.
         HeaderLoc attrLoc(swiftAttr->getLocation());
         diagnose(attrLoc, diag::import_multiple_mainactor_attr,
                  swiftAttr->getAttribute(),
-                 SeenMainActorAttr.getValue()->getAttribute());
+                 seenMainActorAttr.getValue()->getAttribute());
         continue;
       }
 
       if (Type mainActorType = SwiftContext.getMainActorType()) {
         auto typeExpr = TypeExpr::createImplicit(mainActorType, SwiftContext);
         auto attr = CustomAttr::create(SwiftContext, SourceLoc(), typeExpr);
         MappedDecl->getAttrs().add(attr);
-        SeenMainActorAttr = swiftAttr;
+        seenMainActorAttr = swiftAttr;
+      }
+
+      continue;
+    }
+
+    if (isMutabilityAttr(swiftAttr)) {
+      if (seenMutabilityAttr) {
+        const clang::SwiftAttrAttr *nonmutatingAttr =
+            seenMutabilityAttr.getValue()->getAttribute() == "nonmutating"
+                ? seenMutabilityAttr.getValue()
+                : swiftAttr;
+
+        diagnose(HeaderLoc(nonmutatingAttr->getLocation()),
```

Comment:
Looks like this would emit a warning if a nonmutating attribute is specified twice, e.g. `void func() const __attribute__((swift_attr("nonmutating"))) __attribute__((swift_attr("nonmutating")))`.

A more realistic example is:
```cpp
#define NONMUTATING_AND_ANNOTATED1  __attribute__((swift_attr("nonmutating"), annotate("something")))
#define NONMUTATING_AND_ANNOTATED2  __attribute__((swift_attr("nonmutating"), annotate("something_else")))

// ...
void func() const NONMUTATING_AND_ANNOTATED1 NONMUTATING_AND_ANNOTATED2;
```

I think we should only emit a warning if conflicting attributes are specified, WDYT?

Can you please also add a test for this?

Reply:
I think you're right. Should be fixed now, and I've added a test.